### PR TITLE
espnet2 librispeech recipe

### PR DESCRIPTION
**Pull request by sw005320**
Will check the following items in later PRs
- [x] add `README.md`
- [ ] check the STFT analysis condition in detail
- [x] temporally avoid `integration_test_ctc_align_wav.bats` (see https://github.com/espnet/espnet/issues/2146)

- known issues
  - `spm_train` requires huge memory (currently skipping)
  - LM training is very slow (currently skipping) -> fixed in #2114

### CODE REVIEW

**Comment by kamo-naoyuki** [comment]
>   - LM training is very slow (currently skipping)

I heard @ShigekiKarita seq_rnn (= CUDNN-RNN), which is used in espnet2, is slower than RNN-Cell. I don't know why.
Hmm, but very slowness is strange. I don't feel it

**Comment by sw005320** [comment]
> > * LM training is very slow (currently skipping)
> 
> I heard @ShigekiKarita seq_rnn (= CUDNN-RNN), which is used in espnet2, is slower than RNN-Cell. I don't know why.
> Hmm, but very slowness is strange. I don't feel it

This seems to come from some preprocessing before actual training. Probably, some bottleneck due to the huge (but normal range as LM training data) lines of scp files. This is why I’m also testing the scp file splitting but it is also very slow, as I reported in #2106, but I need some time to figure out what happened after I get some extra GPUs.

**Comment by sw005320** [comment]
> I don’t feel it.

Good information.
I guess you already tested it in the librispeech LM training data or similar scale of data.
Then, could you share your config file?

**Comment by kamo-naoyuki** [comment]
The preprocessing includes bpe, so it might be the bottleneck, hmm, but it seems enough fast.
The log file when training, iter_time and train_time, can tell you more information.

**Comment by kamo-naoyuki** [comment]
Note that huge size doesn't affect the training speed. Your problem should be reproduced using small  dataset.
In addition, if the reason is bpe proprocess, it should also make ASR speech degraded. If you see  it in only LM training, it's more strange.

**Comment by sw005320** [comment]
I got the first tentative result.
I think this number is OK by considering that both ASR/LM training is on-going and the beam size is not large.

|dataset|Snt|Wrd|Corr|Sub|Del|Ins|Err|S.Err|
|---|---|---|---|---|---|---|---|---|
|decode_dev_clean_decode_asr_beam_size20_lm_train_lm_bpe_valid.loss.best_asr_model_valid.acc.best|2703|54402|97.2|2.4|0.3|0.3|3.1|35.2|
|decode_dev_other_decode_asr_beam_size20_lm_train_lm_bpe_valid.loss.best_asr_model_valid.acc.best|2864|50948|93.2|6.0|0.8|0.8|7.6|54.7|
|decode_test_clean_decode_asr_beam_size20_lm_train_lm_bpe_valid.loss.best_asr_model_valid.acc.best|2620|52576|97.0|2.6|0.4|0.4|3.4|37.4|
|decode_test_other_decode_asr_beam_size20_lm_train_lm_bpe_valid.loss.best_asr_model_valid.acc.best|2939|52343|92.9|6.1|1.0|0.9|8.0|58.3|

File: egs2/librispeech/asr1/local/data_prep.sh [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+  for chapter_dir in $(find -L $reader_dir/ -mindepth 1 -maxdepth 1 -type d | sort); do
+    chapter=$(basename $chapter_dir)
+    if ! [ "$chapter" -eq "$chapter" ]; then
+      echo "$0: unexpected chapter-subdirectory name $chapter"
+      exit 1
+    fi
+
+    find -L $chapter_dir/ -iname "*.flac" | sort | xargs -I% basename % .flac | \
+      awk -v "dir=$chapter_dir" '{printf "%s %s/%s.flac\n", $0, dir, $0}' >>$wav_scp|| exit 1
+
```
Comment (kamo-naoyuki) [inline egs2/librispeech/asr1/local/data_prep.sh:57]: (Not problem of this PR)
This format violates the format of wav.scp in Kaldi (=flac format is used), so this wav.scp can't perform with feats_type=pitch_fbank, but this is useful for feats_type=raw.

I'll rethink how they can coexist.

**Comment by sw005320** [comment]
@simpleoier, FYI, I just renamed `integration_test_ctc_align_wav.bats` to `fixme_integration_test_ctc_align_wav.bats` to skip the error in this PR, as we discussed in  https://github.com/espnet/espnet/issues/2146.

**Comment by simpleoier** [comment]
> @simpleoier, FYI, I just renamed `integration_test_ctc_align_wav.bats` to `fixme_integration_test_ctc_align_wav.bats` to skip the error in this PR, as we discussed in #2146.

I see. I am fixing it.

**Pull request merged**